glean-core/python/glean/_dispatcher.py

```diff
@@ -105,15 +101,20 @@ def _worker(self):
 
     def _shutdown_thread(self):
         """
-        An atexit handler to tell the worker thread to shutdown and then wait
-        for 1 seconds for it to finish.
+        Tell the worker thread to shutdown and then wait for 1 seconds for it
```

Comment:
In general, when going from mobile to the desktop world, we should try to be a bit more careful at shutdown (which is something we didn't have to pay much attention to, up to now!). We should probably:

- when shutdown is _initiated_, disallow all the recording APIs;
- terminate pending uploads;
- wait for ping I/O (if any).

So, in general, perform a clean telemetry shutdown as we currently do for legacy telemetry on Firefox desktop. This is not something that should be tackled in this bug, but it could be worthwhile filing a bug about this.

Reply:
Yeah -- let's save that for another bug.  It's tricky, because CLI doesn't quite work the same as desktop.